wolfssl/wolfcrypt/sp_int.h

```diff
@@ -95,11 +95,11 @@ extern "C" {
     #error "Size of unsigned int not detected"
 #endif
 
-#if ULONG_MAX == 18446744073709551615UL
+#if ULONG_MAX == W64LIT(18446744073709551615)
     #define SP_ULONG_BITS    64
 
-    typedef unsigned long sp_uint64;
-    typedef          long  sp_int64;
+    typedef word64   sp_uint64;
```

Comment:
Don't do this!
This code is detecting which type has the corresponding bits.
If this isn't working then there is something else wrong.